src/simsopt/geo/boozersurface.py

```diff
@@ -1013,3 +1081,47 @@ def solve_residual_equation_exactly_newton(self, tol=1e-10, maxiter=10, iota=0.,
         self.res = res
         self.need_to_run_code = False
         return res
+
+    def as_dict(self, serial_objs_dict=None) -> dict:
+        d = super().as_dict(serial_objs_dict=serial_objs_dict)
+        
+        # Modify res dict / pop elements that cannot be easily serialized
+        res = self.res
+        if 'gradient' in res.keys():
+            res['gradient'] = res['gradient'].tolist()
+        if 's' in res.keys():
+            res.pop('s')
+        if 'vjp' in res.keys():
+            res.pop('vjp')
+
+        d['res'] = res
+        return d
```

Comment:
can you make it so that when the boozer surface is serialized `need_to_run_code`  is set to `True`?  that way, when things are loaded the code knows that it has to rerun the solver so that the proper 'vjp' is stored in the 'res' dictionary for optimization.

Reply:
Well, I was hoping that the boozer surface wouldn't have to be rerun after loading, unless derivatives are required - then you need to rerun it to get the vjp function. Setting `need_to_run_code` to `True` defeats the purpose of saving the Boozer surface, no?